include/envoy/upstream/cluster_manager.h

```diff
@@ -140,6 +145,16 @@ class ClusterManagerFactory {
   virtual ~ClusterManagerFactory() {}
 
   /**
+   * Allocate a cluster manager from configuration JSON.
+   */
+  virtual ClusterManagerPtr clusterManagerFromJson(const Json::Object& config, Stats::Store& stats,
```

Comment:
Can we remove the other methods from this interface now that we have this? E.g. can `allocateConPool` just be part of `ClusterManagerImpl`.